doc/srv6/srv6-hld-v19.md

```diff
@@ -0,0 +1,552 @@
+
+Schema:
+
+```
+; New table
+; holds prefix to SRv6 SID list encapsulation mapping
+
+key = SRV6_STEER:VRF_NAME:prefix
+                               ; Prefix to be steered
+; field = value
+policy = SRV6_POLICY.key       ; Policy to steer the prefix
+source  = address              ; Source addresses for encapsulation
+
+For example:
+    "SRV6_STEER": {
+        "Vrf-red|11.11.11.0/24": {
+            "policy": "policy1"，
+            "source": "A::1"
+        },
+        "Vrf-blue|2001:a::0/64": {
+            "policy": "policy2"，
+            "source": "A::1"
+        }
+    }
+```
+
+## 2.2 AppDB changes
+
+**New SRV6_SID_LIST_TABLE**
+
+Description: New table to hold SRv6 SID list.
+
+Schema:
+
+```
+; New table
+; holds SRv6 SID list
+
+key = SRV6_SID_LIST_TABLE:segment_name
+
+; field = value
+path = SID,                         ; List of SIDs
+```
+
+**New SRV6_LOCAL_SID_TABLE**
+
+Description: New table to hold local SID to behavior mapping
+
+Schema:
+
+```
+; New table
+; holds local SID to behavior mapping
+
+key = SRV6_LOCAL_SID_TABLE:ipv6address
+
+; field = value
+block_len = blen                   ; bit length of block portion in address, default = 40
+node_len = nlen                    ; bit length of node portion in address, default = 24
+func_len = flen                    ; bit length of function portion in address, default = 16
+arg_len = alen                     ; bit length of argument portion in address, default = 0
+action = behavior                  ; behaviors defined for local SID
+vrf = VRF_TABLE.key                ; VRF name for END.DT46, can be empty
+adj = address,                     ; List of adjacencies for END.X, can be empty
+segment = SRv6_SID_LIST.key,       ; List of segment names for END.B6.ENCAP, can be empty
+source  = address,                 ; List of src addrs for encap for END.B6.ENCAP
+weight  = weight.value,            ; List of assigned weight, can be empty
+```
+
+**Modify ROUTE_TABLE**
+
+Description: Existing Route Table is extended to add SID list. 
+
+Schema:
+
+```
+;Stores a list of routes
+;Status: Mandatory
+
+key = ROUTE_TABLE:VRF_NAME:prefix ;
+nexthop = prefix,              ; IP addresses separated ',' (empty indicates no gateway). May indicate VXLAN endpoint if vni_label is non zero.
+intf = ifindex? PORT_TABLE.key ; zero or more separated by ',' (zero indicates no interface)
+vni_label = VRF.vni            ; zero or more separated by ',' (empty value for non-vxlan next-hops). May carry MPLS label in future.
+router_mac = mac_address       ; zero or more remote router MAC address separated by ',' (empty value for non-vxlan next-hops)
+blackhole = BIT                ; Set to 1 if this route is a blackhole (or null0)
+weight = weight_list           ; List of weights
+sid = SRV6_STEER_TABLE.key,    ; New optional field. List of SID, separated by ',' 
+seg_src = address              ; New optional field. Source addrs for sid encap
+```
+
+## 2.3 Orchestration Agent Changes
```

Comment:
Please have a section to capture examples of kernel routes and kernel interaction

Reply:
done

Reply:
I don't see this addressed. Did you miss any commit?